### ISSUE TITLE
[FindGzOgreNext] Support Ogre 3.x and beyond

### ISSUE THREAD

**Issue by oysstu** [gazebosim/gz-cmake#457]
The ogre-next project has released the next major version ([v3.0.0 Eris](https://github.com/OGRECave/ogre-next/releases/tag/v3.0.0)). It seems that from now on, the project will follow a more conventional semver strategy, where new development is focused on the next major version.

Currently the FindGzOgre2 is hard-coded to find ogre-next releases with major version 2.x.x. It would probably be a good idea to generalize this to FindGzOgreNext. The "USE_NEW_PROJECT_NAME" feature was merged in v2.3.3, so support for v2.3.2 and earlier should probably be left to the existing FindGzOgre2 or dropped completely.

**Comment by j-rivero** [comment]
Wow these are good news! The `FindGzOgreNext` needs a good cleanup and we should try to use plain CMake modules from upstream. I can happily review code changes for this. Thanks @oysstu !

**Comment by oysstu** [comment]
I had a quick look at the ogre-next cmake setup, and it looks like they're not using modern cmake packaging conventions yet. Ideally, they would export the targets and components in a package config in ../share/cmake/. Instead, there's pkg-config files tied together by a big FindOGRE.cmake module in ../lib/../OGRE-Next/cmake.

The way I see it there's two possibilities
a) make the current FindGzOgre functionality work for 3.0.0+. E.g., remove hard checks for version 2 and rename OGRE2 variables to OGRE_NEXT. Remove workarounds for the old-style project layout, if any (e.g. before the move to 'Next').
b) try to rework FindGzOgreNext around the upstream FindOGRE.cmake module instead of utilizing the pkgconfig files directly

The upstream FindOGRE module has been around for a long time though, so I'm sure there's a reason that hasn't been done before.

**Comment by iche033** [comment]
when I wrote the original version using pkg-config many years ago I remember it's because the upstream find ogre module would not always find the correct ogre version or the right ogre plugin libraries. With the new ogre-next name change, things may be different. I think it's a good idea to use the upstream module instead of custom pkg-config code.

**Comment by oysstu** [comment]
I've created an issue in ogre-next (https://github.com/OGRECave/ogre-next/issues/470). Their FindOGRE file still doesn't work well when both ogre-next and ogre is installed. Pkgconfig on the other hand behaves the same even if ogre is installed alongside ogre-next.

I.e., the following prints the same variable definitions even if ogre is installed.
```cmake
cmake_minimum_required(VERSION 3.22)
project(ogre-test)

find_package(PkgConfig REQUIRED)
pkg_check_modules(OGRE_NEXT OGRE-Next REQUIRED)
pkg_check_modules(OGRE_NEXT_HLMS OGRE-Next-Hlms REQUIRED)
pkg_check_modules(OGRE_NEXT_OVERLAY OGRE-Next-Overlay REQUIRED)

get_cmake_property(_variableNames VARIABLES)
list (SORT _variableNames)
foreach (_variableName ${_variableNames})
    message(STATUS "${_variableName}=${${_variableName}}")
endforeach()
```

**Comment by azeey** [comment]
I think it would be great to get Ogre-next 3.0 supported, but I don't think we'd have the bandwidth for getting it in for Jetty. @oysstu Are you interested in helping out?

**Comment by oysstu** [comment]
I might, but I won't have time for the next few weeks. What I suggest is replacing FindOgre2 with FindOgreNext, which only looks for the next-versions, and keep FindOgre for the pre-next projects if backward compatibility is needed.

Is support for ogre classic to be continued? Gz-rendering currently supports v1.9.0, which I suppose was before ogre-next split off into a separate project.

**Comment by mjcarroll** [comment]
> Is support for ogre classic to be continued?

It would be nice to have, as OGRE has progressed to version 14+ now and made many cleanups and optimizations. This is likely a non-trivial update though and I don't believe anyone in the PMC currently has plans to tackle it.

**Comment by j-rivero** [comment]
Closing after #468
